### PR TITLE
add com.autodesk.Eagle

### PR DESCRIPTION
**Pull request by benwaffle**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [ ] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [ ] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [ ] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: com.autodesk.Eagle.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  - --socket=x11
+  - --device=dri
+  - --share=network
+  - --filesystem=home
+modules:
+  - name: eagle
+    buildsystem: simple
+    build-commands:
+      - install -Dm644 com.autodesk.Eagle.desktop /app/share/applications/com.autodesk.Eagle.desktop
+      - install -Dm644 mime.xml /app/share/mime/packages/eagle.xml
```
Comment (hfiguiere) [inline com.autodesk.Eagle.yml:18]:
```suggestion
      - install -Dm644 mime.xml /app/share/mime/packages/${FLATPAK_ID}.xml
```

**Comment by benwaffle** [comment]
bot, build com.autodesk.Eagle

File: mime.xml [hunk 2]
```diff
@@ -0,0 +1,62 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<mime-info xmlns="http://www.freedesktop.org/standards/shared-mime-info">
+	<mime-type type="application/x-eagle-schematic">
+
+		<comment>EAGLE Schematic</comment>
+		<comment xml:lang="de">EAGLE Schaltplan</comment>
+		<comment xml:lang="fr">Schéma électronique EAGLE</comment>
+		<icon name="eagle"/>
```
Comment (hfiguiere) [inline mime.xml:9]: this icon won't be found. At least use the same as for the application.

File: com.autodesk.Eagle.appdata.xml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  </content_rating>
+  <screenshots>
+    <screenshot>
+      https://www.autodesk.com/content/dam/autodesk/www/products/eagle/fy22/images/real-time-design-synchronization-large-1908x1047.jpg
+    </screenshot>
+  </screenshots>
+  <releases>
+    <release version="9.6.2" date="2020-05-27" />
+  </releases>
+</component>
```
Comment (hfiguiere) [inline com.autodesk.Eagle.appdata.xml:27]: it's missing the `project_license`

**Review by hfiguiere — CHANGES_REQUESTED** [review]
you still need to add the flathub.json to limit architectures

**Comment by hfiguiere** [comment]
the appdata file is never installed.

File: com.autodesk.Eagle.yml [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+runtime-version: "21.08"
+sdk: org.freedesktop.Sdk
+command: eagle
+build-options:
+  no-debuginfo: true
+finish-args:
+  - --socket=x11
+  - --device=dri
+  - --share=network
+  - --filesystem=home
```
Comment (hfiguiere) [inline com.autodesk.Eagle.yml:12]: that's a lot of permissions.

File: com.autodesk.Eagle.yml [hunk 5]
```diff
@@ -0,0 +1,47 @@
+app-id: com.autodesk.Eagle
+runtime: org.freedesktop.Platform
+runtime-version: "21.08"
+sdk: org.freedesktop.Sdk
+command: eagle
+build-options:
+  no-debuginfo: true
+finish-args:
+  - --socket=x11
```
Comment (hfiguiere) [inline com.autodesk.Eagle.yml:9]: you also need `--share=ipc`

**Comment by benwaffle** [comment]
bot, build com.autodesk.Eagle

File: com.autodesk.Eagle.yml [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+          type: rotating-url
+          url: https://www.autodesk.com/eagle-download-lin
+      - type: script
+        dest-filename: apply_extra.sh
+        commands:
+          - tar xzf eagle.tar.gz
+          - mv eagle-*/* .
+          # Remove provided X libraries. They cause issues with current mesa and the system libraries seem to work fine.
+          - rm lib/libx* lib/libX*
+          - install -Dm644 doc/eagle.1 export/share/man/man1/eagle.1
```
Comment (hfiguiere) [inline com.autodesk.Eagle.yml:41]:
```suggestion
          - install -Dm644 doc/eagle.1 export/share/man/man1/eagle.1
```

Nobody will see the manpage.

File: com.autodesk.Eagle.desktop [hunk 7]
```diff
@@ -0,0 +1,9 @@
+[Desktop Entry]
+Version=1.0
+Type=Application
+Name=Autodesk EAGLE
+TryExec=/app/eagle
```
Comment (hfiguiere) [inline com.autodesk.Eagle.desktop:5]:
```suggestion
```

**Comment by benwaffle** [comment]
bot, build com.autodesk.Eagle

**Comment by benwaffle** [comment]
bot, build com.autodesk.Eagle

**Comment by tinywrkb** [comment]
The bundled ngspice is quite outdated. IIRC Eagle allows changing the location of the simulator in its settings.  
Please consider packaging an updated ngspice version, so the user will have the option to try a more recent simulator.  
I have a ngspice module packaged [here](https://github.com/tinywrkb/flatpaks/blob/master/shared-modules/ngspice/ngspice.json), though I'm not sure if only the simulator is enabled or also other tools.

**Comment by benwaffle** [comment]
I think we should try to do that after this is merged.

**Comment by barthalion** [comment]
bot, build com.autodesk.Eagle

File: com.autodesk.Eagle.appdata.xml [hunk 8]
```diff
@@ -0,0 +1,28 @@
+<?xml version="1.0" encoding="utf-8"?>
+<component type="desktop-application">
+  <id>com.autodesk.Eagle</id>
+  <name>Autodesk Eagle</name>
+  <summary>Powerful suite for schematic capture and printed circuit board design (aka eaglecad)</summary>
+  <description>
```
Comment (barthalion) [inline com.autodesk.Eagle.appdata.xml:6]: The description should contain a disclaimer that it's not an officially maintained app (see commercial Jetbrains applications) and mention it requires an account to work.

**Pull request closed without merging**